### PR TITLE
[SPIR-V] Update SPV_INTEL_function_pointers extension

### PR DESCRIPTION
**Pull request by zuban32**
Replace OpFunctionPointerINTEL with a constant one

### CODE REVIEW

File: sycl/doc/extensions/SPIRV/SPV_INTEL_function_pointers.asciidoc [hunk 1]
```diff
@@ -207,25 +209,28 @@ Modify Section 3.32.6, Type-Declaration Instructions, change the third sentence
...
+a composite constant or stored somewhere for further usage
+in *OpFunctionPointerCallINTEL*. +
  +
 _Result Type_ must be an *OpTypePointer*. Its _Type_ operand must be the same
 *OpTypeFunction* which was used as _Function Type_ operand of the _Function_
 operand. Its _Storage Class_ operand must be *CodeSectionINTEL*
 | <<Capability,Capability>>: +
 *FunctionPointersINTEL*
-| 4 | 5600 | '<id>' 'Result Type' | '<id> Result ' | '<id>' 'Function'
+| 4 | ?? | '<id>' 'Result Type' | '<id> Result ' | '<id>' 'Function'
```
Comment (AlexeySachkov) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_function_pointers.asciidoc:229]: This change should be reverted, I guess

Comment (zuban32) [reply sycl/doc/extensions/SPIRV/SPV_INTEL_function_pointers.asciidoc]: Fixed

**Comment by AlexeySachkov** [comment]
@intel/dpcpp-specification-reviewers, any feedback?

**Comment by bader** [comment]
@intel/dpcpp-specification-reviewers, please, take a look.

**Comment by bader** [comment]
@zuban32, could you resolve merge conflicts, please?

**Comment by bader** [comment]
> @zuban32, could you resolve merge conflicts, please?

@zuban32, ping.

**Review by AlexeySachkov — CHANGES_REQUESTED** [review]
This PR should be superseded in favor of #4483 per my [comment](https://github.com/intel/llvm/pull/4883#issuecomment-1031354626) there

**Pull request closed without merging**